src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -468,17 +484,45 @@ private BoundExpression MakeConversionNode(BoundExpression rewrittenOperand, Typ
                      rewrittenOperand.Type.SpecialType != SpecialType.System_DateTime)
                 {
                     // error CS0029: Cannot implicitly convert type '{0}' to '{1}'
-                    _diagnostics.Add(
+                    diagnostics.Add(
                         ErrorCode.ERR_NoImplicitConv,
                         rewrittenOperand.Syntax.Location,
                         rewrittenOperand.Type,
                         rewrittenType);
                 }
+            }
 
-                return _factory.NullOrDefault(rewrittenType);
+            return conversion;
+        } 
+
+        private static BoundExpression MakeConversionNodeIfNecessaryWithoutLowering(
+            BoundExpression operand, 
+            TypeSymbol type, 
+            SyntaxNode syntax, 
+            CSharpCompilation compilation, 
+            DiagnosticBag diagnostics, 
+            bool @checked, 
+            bool acceptFailingConversion = false)
+        {
+            if (operand.Type.Equals(type))
+            {
+                return operand;
             }
+            Conversion conversion = MakeConversion(operand, type, compilation, diagnostics, acceptFailingConversion); 
 
-            return MakeConversionNode(rewrittenOperand.Syntax, rewrittenOperand, conversion, rewrittenType, @checked);
+            if (!conversion.IsValid)
+            {
+                return SyntheticBoundNodeFactory.NullOrDefault(type, syntax);
```

Comment:
Hmm, I think in the simple cases such as 
```cs
    void M1()
    {
        M2(1);
    }
    void M2(string x )
    { }
```
we can't even get this far, it would have stopped before calling into `MakeArgumentsInEvaluationOrder` because `HasErrors` would be `true` for the bound node? In what case we can reach line?